core/src/main/java/org/apache/carbondata/core/metadata/CarbonMetadata.java

```diff
@@ -143,7 +143,7 @@ public CarbonDimension getCarbonDimensionBasedOnColIdentifier(CarbonTable carbon
     List<CarbonDimension> listOfCarbonDims =
         carbonTable.getDimensionByTableName(carbonTable.getTableName());
     for (CarbonDimension dimension : listOfCarbonDims) {
-      if (dimension.getColumnId().equals(columnIdentifier)) {
+      if (dimension.getColumnId().equalsIgnoreCase(columnIdentifier)) {
```

Comment:
Will the comparison for columnId still need case insensitive?
If it is yes, I'd suggest you to leave comments for this (for old store before version XXX).

Reply:
this is the case for non-transactional flow. It uses column name for columnid as well. It is the safest way to do the check and there would not be any side effects on it,